src/server.js

```diff
@@ -0,0 +1,150 @@
+"use strict";
+var http = require('http');
+var graphql = require('graphql');
+require('babel-register');
+var WebSocketServer = require('websocket').server;
+var graphql_tools = require('graphql-tools');
+var graphql_validator = require('graphql/validation');
+var graphql_execution = require('graphql/execution');
+
+class Server {
+  /*
+  options {
+    schema: GraphQLSchema
+    contextValue?: any,
+    rootValue?: any,
+    formatResponse?: (Object) => Object,
+    validationRules?: Array<any> 
+    triggerGenerator?: (Object) => [Object]
+  }
+  */
+  constructor(options, httpServer) {
+    this.options = options;
+    this.triggers = {}; // trigger_object: [{connection, sub_id}]
+    //initialize http server
+
+    //init and connect websocket server to http
+    this.wsServer = new WebSocketServer({
+      httpServer: httpServer,
+      autoAcceptConnections: false
```

Comment:
look at other options to websocket server, like ignoreXForwardedFor. Do we need them?